src/Tribe/Events/Attendee_List_Display.php

```diff
@@ -0,0 +1,223 @@
+			 * with it according to the >> caller << context.
+			 */
+			throw new InvalidArgumentException( '$post should be either a post ID or a WP_Post object.' );
+		}
+
+		return (bool) Tribe__Tickets_Plus__Attendees_List::is_hidden_on( $post );
+	}
+
+	/**
+	 * @param bool $should_hide Whether the optout form should be hidden or not.
+	 * @filter tribe_tickets_plus_hide_attendees_list_optout 10 1
+	 * @see    \Tribe\Tickets\Events\Events_Service_Provider::hooks
+	 *
+	 * @return bool
+	 */
+	public function should_hide_optout( $should_hide ) {
+		try {
+			global $post;
+
+			return $this->is_event_hiding_attendee_list( $post );
+		} catch ( Exception $e ) {
+			// Eg: global $post not a WP_Post object.
+			return $should_hide;
+		}
+	}
+
+	/**
+	 * Determines whether this post is displaying the Attendees List.
+	 *
+	 * @param int|WP_Post $post The Post being checked.
+	 * @action save_post_tribe_events 10 1
+	 * @see    \Tribe\Tickets\Events\Events_Service_Provider::hooks
+	 *
+	 * @retyrn void
+	 */
+	public function maybe_update_attendee_list_hide_meta( $post ) {
+		// Attendees list is a Plus feature, if ET Plus is not present we don't have to update the meta.
+		if ( ! class_exists( 'Tribe__Tickets_Plus__Main' ) ) {
+			return;
+		}
+
+		// Early bail: is an autosave or auto-draft.
+		if ( wp_is_post_autosave( $post ) || wp_is_post_revision( $post ) ) {
+			return;
+		}
+
+		if ( ! $post instanceof WP_Post ) {
+			$post = get_post( $post );
+		}
+
+		// Early bail: Invalid post.
+		if ( ! $post instanceof WP_Post ) {
+			return;
+		}
+
+		$post_types_allowed_to_have_tickets = (array) tribe_get_option( 'ticket-enabled-post-types', [] );
+
+		// Early bail: This post type can't have tickets.
+		if ( ! in_array( $post->post_type, $post_types_allowed_to_have_tickets ) ) {
+			return;
+		}
+
+		$this->track_shortcode_driven_meta( $post );
+
+		if ( $this->is_using_blocks() ) {
+			$is_showing_attendee_list = $this->is_showing_attendee_list_with_blocks( $post );
+		} else {
+			$is_showing_attendee_list = $this->is_showing_attendee_list_with_classical_editor( $post );
+		}
+
+		/**
+		 * Returns true if the post is displaying a list of attendees.
+		 *
+		 * You can use this filter to let the system know that you're displaying
+		 * the Attendee List in some other way.
+		 *
+		 * @since 4.12.0
+		 *
+		 * @param bool    $is_showing_attendee_list Whether the post is showing the attendee list or not
+		 * @param WP_Post $post                     The WP_Post object being checked
+		 * @param bool    $is_using_blocks          Whether the post is using Blocks or not
+		 *
+		 * @return bool
+		 */
+		$is_showing_attendee_list = (bool) apply_filters(
+			'tribe_tickets_event_is_showing_attendee_list',
+			$is_showing_attendee_list,
+			$post,
+			$this->is_using_blocks()
+		);
+
+		update_post_meta( $post->ID, Tribe__Tickets_Plus__Attendees_List::HIDE_META_KEY, $is_showing_attendee_list );
```

Comment:
Same as the `is_hidden_on()` method, we should create this here, and eventually have that in ET+ from inheritance behavior.

Reply:
Also, let's return the contents of `update_post_meta` should be useful for tests.w

Reply:
Thanks. I've added the return of `update_post_meta` in this method https://github.com/moderntribe/event-tickets/pull/1633/commits/e1ef40e53fd3871c40125703f98c00c9e26d24ce